src/main/java/kotlinx/reflect/lite/KClass.kt

```diff
@@ -0,0 +1,131 @@
+package kotlinx.reflect.lite
+
+import kotlinx.reflect.lite.descriptors.*
+import kotlinx.reflect.lite.descriptors.ClassDescriptor
+
+/**
+ * Represents a class and provides introspection capabilities.
+ *
+ * @param T the type of the class.
+ */
+public interface KClass<out T> : KDeclarationContainer, KAnnotatedElement, KClassifier {
```

Comment:
Let's make all these interfaces `sealed`. It doesn't make sense for anyone else to implement them.

Reply:
Alas, it cannot be sealed due to being in another package. We'll figure something out later